lib/elements/analysis_results_controller.dart

```diff
@@ -118,3 +178,15 @@ class AnalysisResultsController {
     toggle.text = _hideMsg;
   }
 }
+
+class LineInfo {
```

Comment:
I'm kind of surprised there isn't already a class with these properties in the codebase, but I can't find one. There's `Position` in `/lib/editing/editor.dart`, but it doesn't include length.

Reply:
Or, given that `Position` is identical in structure to the interface `Position` from LSP, consider using the LSP interface [`Range`](https://microsoft.github.io/language-server-protocol/specification#range) instead. Although it might not be as convenient for your purposes.

Reply:
The [Location](https://pub.dev/documentation/analysis_server_lib/latest/analysis_server_lib/Location-class.html) type is used in AnalysisError and DiagnosticMessage classes, so `dart-services` adds the line, charStart, and charLength to an AnalysisIssue protocol buffer message that the client gets. The client doesn't actually use the analysis_server_lib package at the moment.

We could add two new `Postition` and `Range` messages to our protocol buffer, but that might be out of scope here.